soc/arm/nordic_nrf/Kconfig.peripherals

```diff
@@ -4,305 +4,298 @@
 # SPDX-License-Identifier: Apache-2.0
 
 config HAS_HW_NRF_ACL
```

Comment:
Are we keeping these for compatibility, or should they be removed?

Reply:
We are keeping them, as they are useful for creating Kconfig dependencies for drivers or other modules that need particular hardware or feature.

Reply:
But can't they use `$(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF_ACL))` instead?
Now we have 2 ways to do the same thing, I get that this removes the pattern for having select, but I don't really understand why you would do this change for example:
```
config NFCT_PINS_AS_GPIOS
	bool "NFCT pins as GPIOs"
-	depends on HAS_HW_NRF_NFCT
+	depends on $(dt_has_compat,$(DT_COMPAT_NORDIC_NRF_NFCT))
```

Reply:
> But can't they use `$(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF_ACL))` instead?

They could. `HAS_HW_NRF_*` are just more convenient to use. Plus you can refer to `CONFIG_HAS_HW_NRF_*` definitions from source code (we've got a few such uses in tree).

> Now we have 2 ways to do the same thing, I get that this removes the pattern for having select, but I don't really understand why you would do this change for example:
> [..]

`HAS_HW_NRF_NFCT` is set to "y" only when a `"nordic,nrf-nfct"` compatible node is enabled, and the `NFCT_PINS_AS_GPIOS` option needs to be available when such node just exists (i.e. when a given SoC supports NFCT). That's why I needed to do this change.

Reply:
Ok, thank you for the clarification. I see now that you have actually changed the semantics of the `CONFIG_HAS_HW_NRF_*` from the HW being supported by the SoC to the HW being enabled in devicetree.

So `$(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF_ACL))` represents the old semantics, then, correct?

Reply:
> So `$(dt_compat_enabled,$(DT_COMPAT_NORDIC_NRF_ACL))` represents the old semantics, then, correct?

No. That would need to be `$(dt_has_compat,$(DT_COMPAT_NORDIC_NRF_ACL))`. Such dependencies are now used in `modules/hal_nordic/nrfx/Kconfig` instead of `HAS_HW_NRF_*` symbols used previously, because nrfx drivers need to be available when corresponding HW is supported by a given SoC, not only when it is enabled to be used by Zephyr drivers.